diff --git "a/.pio\\libdeps\\i2cHelper-esp32thing_plus\\Ethernet\\src\\Ethernet.h" "b/patches\\Ethernet.h"
index 376e6c5..d124d2b 100644
--- "a/.pio\\libdeps\\i2cHelper-esp32thing_plus\\Ethernet\\src\\Ethernet.h"
+++ "b/patches\\Ethernet.h"
@@ -219,6 +219,20 @@ public:
 	uint8_t status();
 	virtual int connect(IPAddress ip, uint16_t port);
 	virtual int connect(const char *host, uint16_t port);
+#ifdef ESP32
+	// See https://github.com/espressif/arduino-esp32/issues/2704
+	// and https://github.com/arduino-libraries/Ethernet/pull/107/files	
+	int connect(IPAddress ip, uint16_t port, int timeout)
+	{
+		connect(ip, port);
+		setConnectionTimeout(timeout);
+	}
+	int connect(const char* host, uint16_t port, int timeout)
+	{
+		connect(host, port);
+		setConnectionTimeout(timeout);
+	}
+#endif
 	virtual int availableForWrite(void);
 	virtual size_t write(uint8_t);
 	virtual size_t write(const uint8_t *buf, size_t size);
@@ -258,6 +272,14 @@ public:
 	EthernetClient available();
 	EthernetClient accept();
 	virtual void begin();
+#ifdef ESP32
+	// See https://github.com/espressif/arduino-esp32/issues/2704
+	// and https://github.com/arduino-libraries/Ethernet/pull/107/files
+	void begin(uint16_t port)
+	{
+		_port = port;
+	}
+#endif
 	virtual size_t write(uint8_t);
 	virtual size_t write(const uint8_t *buf, size_t size);
 	virtual operator bool();
